cmd/minishift/cmd/start.go

```diff
@@ -179,7 +183,18 @@ func runStart(cmd *cobra.Command, args []string) {
 		atexit.Exit(1)
 	}
 
-	clusterUp(&config)
+	oc, err := clusterUp(&config)
+	if err != nil {
+		glog.Errorln("Error starting the cluster: ", err)
+		atexit.Exit(1)
+	}
+
+	// Update minishift-data.json for oc path
+	minishiftData, err := minishiftdata.Create(machinesDirPath)
+	minishiftData.OcPath = filepath.Join(oc.GetCacheFilepath(), constants.OC_BINARY_NAME)
+	if err = minishiftdata.WriteMinishiftData(minishiftData); err != nil {
```

Comment:
This does not encapsulate data and logic... makes it hard to actually use the functionality.